### ISSUE TITLE
Issues when running CI/CD via recent vagrant image

### ISSUE THREAD

**Issue by garyhodgson** [fabric8io/default-jenkins-dsl#2]
Hi,

I was playing around with the CI/CD stuff and found a couple of issues.  

The -it job failed because the branch was missing in gogs.   It seems that the postBuildSteps from the seed job were somehow not correctly applied - because the step to push the build branch to gogs was completely missing.

After manually adding these steps I then hit another problem:

```
21:35:41  > git rev-parse ${TAG-PREFIX}-1.0.15^{commit} # timeout=10
21:35:41 ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
```

This appears to be due to the Branch specifier being defined in the -it and -dev-deploy jobs as being "${TAG-PREFIX}-${RELEASE_NUMBER}" which refers to env variables, I believe these should be "$TAG-PREFIX-$RELEASE_NUMBER" as they are parameters.  At least the -it job worked when I manually changed them.

Cheers,
Gary

**Comment by jstrachan** [comment]
great catch thanks @garyhodgson. Sorry for the delay replying!

This just came up on the mailing list too:
https://groups.google.com/forum/#!topic/fabric8/ldIMGhRybTE

I agree its broken. We're just in the process of merging some changes from here: https://github.com/christian-posta/default-jenkins-dsl which might work a bit better. That one uses gerrit code review too.

I hope we can get it working so OOTB everything just works; then longer term create a reusable library of CI / CD workflows folks can choose between; or can fork or write their own etc

**Comment by jstrachan** [comment]
@garyhodgson if you got it working locally do you wanna just submit a Pull Request with your changes? That'd be awesome!

**Comment by christian-posta** [comment]
hey @garyhodgson thanks for trying it out, and so sorry for the delay. i'm about to commit the merge of my changes to this repo; as it turns out, i'm going to take out the git branch creation stuff for now as it causes a cycle of builds with the gogs web hook (notices a new branch created so kicks off a new build... which creates a new branch, etc). we'll need to work out what gogs should do when new branches are created (i.e., have more fine grained web hooks) and also allow users to configure at what step in the process a branch should be tagged (i.e., at the initial build step? after a build has been deployed to QA? after a canary release has been promoted to rolling upgrade in prod?)

**Comment by pulse00** [comment]
@garyhodgson i've also tried your fork locally, everything is working (if one starts the build manually). The build assumes a kubernetes namespace named `development` though, which doesn't exist by default. But as long as you create it, the builds work fine.

**Comment by christian-posta** [comment]
Right, in our demo we kicked off the build manually. We did this for two
reasons:

for a demo, we didn't want all of the magic happening at once, so we slowed
things down and manually kicked off the build.

We also disabled the gogs webhook for the demo because we wanted to demo
tagging/releasing artifacts into nexus and aligning those with the gogs
repo. That will need some work though since the gogs webhook treats all
commits to the repo the same vs tags/branches differently.

I've just changed it to remove the git branching stuff for now while we
hack out the jenkins workflow stuff and fabric8.yaml .. and make it
configurable where/when to do that.

On Thu, Jul 2, 2015 at 11:54 AM, Robert Gründler notifications@github.com
wrote:

> @garyhodgson https://github.com/garyhodgson i've also tried your fork
> locally, everything is working fine (if one starts the build manually). The
> build assumes a kubernetes namespace named development though, which
> doesn't exist by default. But as long as you create it, the builds work
> fine.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fabric8io/default-jenkins-dsl/issues/2#issuecomment-118126409
> .

## 

_Christian Posta_
twitter: @christianposta
http://www.christianposta.com/blog
http://fabric8.io

**Comment by jstrachan** [comment]
maybe we should use an env var for now so folks can enable/disable the 'build on create' option? I like the idea of it building by default BTW :)

**Comment by jstrachan** [comment]
it can then be an OpenShift Template parameter whether to enable/disable this feature when you run the cdelivery app

**Comment by christian-posta** [comment]
On Thu, Jul 2, 2015 at 12:08 PM, James Strachan notifications@github.com
wrote:

> maybe we should use an env var for now so folks can enable/disable the
> 'build on create' option?

is that for the openshift build thingy?
it's enabled by default :)
https://github.com/fabric8io/default-jenkins-dsl/blob/master/seedBuilds.groovy#L261
I've removed the branching stuff for now

> I like the idea of it building by default BTW :)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fabric8io/default-jenkins-dsl/issues/2#issuecomment-118134311
> .

## 

_Christian Posta_
twitter: @christianposta
http://www.christianposta.com/blog
http://fabric8.io

**Comment by jstrachan** [comment]
ah cool :) ignore me!

**Comment by jstrachan** [comment]
I guess if you wanted to disable it - say for a demo - we could make the enable/disable conditional on an env var - and then add an env var / template parameter to the jenkins kube json

**Comment by christian-posta** [comment]
Yah that sounds good too... i'll start poking around what you've done for
the fabric8.yaml stuff as i think that's gonna be the road to take for this
kind of thing. with a library of CI/CD steps/configurations, and the
fabric8.yaml file, we should be able to control this kind of thing from a
project. will be hawt :)

On Thu, Jul 2, 2015 at 12:17 PM, James Strachan notifications@github.com
wrote:

> I guess if you wanted to disable it - say for a demo - we could make the
> enable/disable conditional on an env var - and then add an env var /
> template parameter to the jenkins kube json
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fabric8io/default-jenkins-dsl/issues/2#issuecomment-118138559
> .

## 

_Christian Posta_
twitter: @christianposta
http://www.christianposta.com/blog
http://fabric8.io

**Comment by jstrachan** [comment]
BTW master of fabric8 now generates a fabric8.yml (with a wizard page in the create project UI) like this:
https://github.com/fabric8io/fabric8/blob/master/components/fabric8-devops/src/test/resources/fabric8.yml#L4

if `codeReview` is true then I figure we should generate a Gerrit project (if Gerrit is available in kubernetes).

I was thinking we'd rename the fabric8:create-build-config to be fabric8:devops (so its more clear) - this mvn plugin already loads the fabric8.yml and do stuff based on its content
https://github.com/jstrachan/fabric8/blob/cdelivery/fabric8-maven-plugin/src/main/java/io/fabric8/maven/CreateBuildConfigMojo.java#L314-314

so we can move the generation of the gerrit stuff into that mvn plugin and it'll happen automatically based on the contents of the fabric8.yml ;)

**Comment by jstrachan** [comment]
we'll cut a release soon with the fabric8.yml code inside so folks can play with it
